Add R2DBC support (#2545)
testcontainers/testcontainers-java: docs/modules/databases/r2dbc.md

# R2DBC support

You can obtain a temporary database in one of two ways:

 * **Using a specially modified R2DBC URL**: after making a very simple modification to your system's R2DBC URL string, Testcontainers will provide a disposable stand-in database that can be used without requiring modification to your application code.
 * **JUnit @Rule/@ClassRule**: this mode starts a database inside a container before your tests and tears it down afterwards.

## Database containers launched via R2DBC URL scheme

As long as you have Testcontainers and the appropriate R2DBC driver on your classpath, you can simply modify regular R2DBC connection URLs to get a fresh containerized instance of the database each time your application starts up.

> Started container will be terminated when the `ConnectionFactory` is closed.

Reviewer: ```suggestion
The started container will be terminated when the `ConnectionFactory` is closed.
```

Author: I knew it! I freaking knew it!! 😂

!!! warning

>     both the database module like `org.testcontainers:mysql` **and** `org.testcontainers:r2dbc` need to be on your application's classpath at runtime for this to work.

Reviewer: ```suggestion
    Both the database module (e.g. `org.testcontainers:mysql`) **and** `org.testcontainers:r2dbc` need to be on your application's classpath at runtime.
```

**Original URL**: `r2dbc:mysql://localhost:3306/databasename`

1. Insert `tc:` after `r2dbc:` as follows. Note that the hostname, port and database name will be ignored; you can leave these as-is or set them to any value.

> 1. Specify the mandatory Docker tag of database's official image that you want to use as `TC_IMAGE_TAG`

Reviewer: ```suggestion
1. Specify the mandatory Docker tag of database's official image that you want to using a `TC_IMAGE_TAG` query parameter.

**Note that, unlike Testcontainers' JDBC URL support, it is not possible to specify an image tag in the 'scheme' part of the URL, and it is always necessary to specify a tag using `TC_IMAGE_TAG`.**
```

Author: > want to using a

it _feels_ that a word missing between `to` and `using`. Don't we miss something here?

Reviewer: Oops!

Reviewer: Fixed my suggestion.
